build.ps1

```diff
@@ -1,19 +1,67 @@
-# script to create the final package in out\platyPS
-
+<#
+.SYNOPSIS
+    Builds the MarkDown/MAML DLL and assembles the final package in out\platyPS.
+#>
+[CmdletBinding()]
 param(
     [ValidateSet('Debug', 'Release')]
     $Configuration = "Debug",
     [switch]$SkipDocs
 )
 
 # build .dll
-# msbuild is part of .NET Framework, we can try to get it from well-known location.
-if (-not (Get-Command -Name msbuild -ErrorAction Ignore)) {
-    Write-Warning "Appending probable msbuild path"
-    $env:path += ";${env:SystemRoot}\Microsoft.Net\Framework\v4.0.30319"
+[string] $msbuildPath = $null
+
+$msbuildCmd = Get-Command -Name msbuild -ErrorAction Ignore
+
+if ($msbuildCmd) {
+    $msbuildPath = $msbuildCmd.Path
+} else {
+    Write-Warning 'Searching for msbuild'
+
+    # For more info on vswhere.exe:
+    #    https://blogs.msdn.microsoft.com/heaths/2017/02/25/vswhere-available/
+    #    https://github.com/Microsoft/vswhere/wiki/Find-MSBuild
+    $vswherePath = "${env:ProgramFiles(x86)}\Microsoft Visual Studio\Installer\vswhere.exe"
+    if (Test-Path $vswherePath) {
+
+        $vsInstallPath = & $vswherePath -latest `
+                                        -requires Microsoft.VisualStudio.Component.Roslyn.Compiler, Microsoft.Component.MSBuild `
+                                        -property installationPath
+
+        if ($? -and $vsInstallPath) {
```

Comment:
can we, please, avoid using `$?`
it's a convenient to use in interactive sessions, scripts are harder to read because of it.